src/main/java/net/dv8tion/jda/api/entities/Guild.java

```diff
@@ -1234,7 +1234,7 @@ default List<Member> getMembersByEffectiveName(@Nonnull String name, boolean ign
     @Nonnull
     default List<Member> getMembersWithRoles(@Nonnull Role... roles)
     {
-        Checks.noneNull(roles, "Roles");
+        Checks.notNull(roles, "Roles");
         return getMembersWithRoles(Arrays.asList(roles));
     }
```

Comment:
No, you got me wrong. As said before, it can't be null due to two methods having the same name and same number of parameters. The compiler would simply not allow it without explicit casting. And if somebody does that, it's more or less their fault imo.

And no need to check for `nonNull` again since you already do that in `getMembersWithRoles(List<Role>)`.

Reply:
That method has to check `notNull` otherwise something like `Role[] roles = null; guild.getMembersWithRoles(roles);` would cause `Arrays#asList` to throw an NPE

> And no need to check for `noneNull` again since you already do that in `getMembersWithRoles(List<Role>)`

Yes

Reply:
> That method has to check notNull otherwise something like Role[] roles = null; guild.getMembersWithRoles(roles); would cause Arrays#asList to throw an NPE

Yes and no. As said, you would explicitly need to cast it / define it like that so it compiles. And as a user you should see this is simply wrong. So imo it's fine to let the user run into this. Because essentially you just trade a `NullPointerException` with an `IllegalArgumentException`. Doesn't help much but needs more code. So I would need to remove it.

From what I see it's also not used in other places and consistency is more important. So imo, it's really fine to remove that check again.